src/common/Gateway.sol

```diff
@@ -101,9 +101,7 @@ contract Gateway is Auth, Recoverable, IGateway {
             bytes memory message = remaining.slice(0, length);
             remaining = remaining.slice(length, remaining.length - length);
 
-            uint256 executionGas = gasService.messageGasLimit(localCentrifugeId, message);
-            require(gasleft() >= executionGas + GAS_FAIL_MESSAGE_STORAGE, NotEnoughGasToProcess());
-
+            require(gasleft() > GAS_FAIL_MESSAGE_STORAGE, NotEnoughGasToProcess());
```

Comment:
Unrelated note: do we still need the internal `_process` method? Think I asked this in a previous PR, but it's now even simpler.

Reply:
It's what is done now, called `GatewatExt` (probably should rename it): https://github.com/centrifuge/protocol/blob/main/src/common/Gateway.sol

Reply:
Done [1cd1834](https://github.com/centrifuge/protocol/pull/646/commits/1cd1834303a4233b167a9208705d5e94bc5a961c)